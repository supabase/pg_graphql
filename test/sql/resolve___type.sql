begin;

    create table account(
        id serial primary key,
        email varchar(255) not null,
        encrypted_password varchar(255) not null,
        created_at timestamp not null,
        updated_at timestamp not null
    );


    select jsonb_pretty(
        gql.resolve($$
        {
          __type(name: "Account") {
            kind
            fields {
                name
            }
          }
        }
        $$)
    );

rollback;
