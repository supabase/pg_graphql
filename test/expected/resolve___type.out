begin;
    create table account(
        id serial primary key,
        email varchar(255) not null,
        encrypted_password varchar(255) not null,
        created_at timestamp not null,
        updated_at timestamp not null
    );
    select jsonb_pretty(
        gql.resolve($$
        {
          __type(name: "Account") {
            kind
            fields {
                name
            }
          }
        }
        $$)
    );
                  jsonb_pretty                   
-------------------------------------------------
 {                                              +
     "data": {                                  +
         "__type": {                            +
             "kind": "OBJECT",                  +
             "fields": [                        +
                 {                              +
                     "name": "nodeId"           +
                 },                             +
                 {                              +
                     "name": "id"               +
                 },                             +
                 {                              +
                     "name": "email"            +
                 },                             +
                 {                              +
                     "name": "encryptedPassword"+
                 },                             +
                 {                              +
                     "name": "createdAt"        +
                 },                             +
                 {                              +
                     "name": "updatedAt"        +
                 }                              +
             ]                                  +
         }                                      +
     },                                         +
     "errors": [                                +
     ]                                          +
 }
(1 row)

rollback;
